### ISSUE TITLE
Ui got freez when the child have many items 

### ISSUE THREAD

**Issue by kisonix** [ikws4/TreeView#3]
https://user-images.githubusercontent.com/109852917/181875388-b53b65d7-64e4-4434-af9b-b4f6dc81d0c0.mp4

vendor (include 40 folders and 1 files as child)
node_modules (include 300+ folders as child)

I think it's occured by [this](https://github.com/ikws4/TreeView/blob/main/app/src/main/java/io/github/ikws4/treeview/documentfile/DocumentFileTreeAdapter.java#L70)  or maybe [this](https://github.com/ikws4/TreeView/blob/main/app/src/main/java/io/github/ikws4/treeview/documentfile/DocumentFileTreeAdapter.java#L85) 

I have tried to used with `java.util.concurrent.Executor` (in case asynctask is deprecated and not familiar with rxjava) but somehow the view didn't show anything like it doesn't get the notify

**Comment by ikws4** [comment]
Yeah, I think you're right `DocumentFile.listFiles()` is pretty slow compare with `File.listFiles()` according  to [this](https://stackoverflow.com/a/47367809), maybe you can try his approach to see if you can get better performance?

**Comment by kisonix** [comment]
> Yeah, I think you're right `DocumentFile.listFiles()` is pretty slow compare with `File.listFiles()` according to [this](https://stackoverflow.com/a/47367809), maybe you can try his approach to see if you can get better performance?

I'm stuck I tried [this](https://stackoverflow.com/questions/42186820/why-is-documentfile-so-slow-and-what-should-i-use-instead)  but some of the constructor are not public, I have no idea to make it:)

**Comment by ikws4** [comment]
Check out [ItzNotABug/DocumentFileCompat](https://github.com/ItzNotABug/DocumentFileCompat), it is much faster than `DocumentFile` and the usage just the same. 

See https://github.com/ikws4/TreeView/commit/1c5b98f25fea5cd700bef73f965b951752133397.

**Comment by kisonix** [comment]
last commit was resolved the issues above but I got new bugs 

1. Somehow the view always recognize every child as expandable I think that's why [this](https://github.com/ikws4/TreeView/blob/main/treeview/src/main/java/io/github/ikws4/treeview/TreeView.java#L75) code doesn't work

2. sorting doesn't apply

3. I think it caught by (1) unable to click item which are not expandable (In my app I have implemented debug activity and it saying `java.lang.nullpointerexception attempt to get length of null array `

[demo video](https://mi-link.herokuapp.com/dl/0/AgAD-AYAAihnKVc.mp4?st=1)

**Comment by ikws4** [comment]
> Somehow the view always recognize every child as expandable I think that's why [this](https://github.com/ikws4/TreeView/blob/main/treeview/src/main/java/io/github/ikws4/treeview/TreeView.java#L75) code doesn't work

It has nothing to do with the library itself, but the issue with [ItzNotABug/DocumentFileCompat](https://github.com/ItzNotABug/DocumentFileCompat), it's `isDirectory` seem not working. So I changed it to `!isFile()`.

> sorting doesn't apply

Sorting does applied, but may not as you wanted.
<img width="591" alt="Screen Shot 2022-07-30 at 7 38 38 PM" src="https://user-images.githubusercontent.com/47056144/181909201-8f007c14-c9b7-41ef-a534-82e486a74bb3.png">

So I changed to this
<img width="738" alt="Screen Shot 2022-07-30 at 7 43 56 PM" src="https://user-images.githubusercontent.com/47056144/181909395-72f125c0-5930-43cb-be81-2279f065fb0f.png">
